src/features/projectsV2/ProjectDetails/index.tsx

```diff
@@ -104,18 +111,46 @@ const ProjectDetails = ({
   // add  project site query
   useEffect(() => {
     const projectSites = singleProject?.sites;
-    if (!projectSites || !projectSites[selectedSite]) {
+    if (!projectSites) {
       return;
     }
-    const newSiteId = projectSites[selectedSite].properties.id;
-    const pathname = `/${locale}/prd/${singleProject.slug}`;
+    if (!router.isReady) return;
+    const pushWithShallow = (pathname: string, queryParams = {}) => {
+      router.push({ pathname, query: queryParams }, undefined, {
+        shallow: true,
+      });
+    };
 
-    const query = updateUrlWithParams(router.asPath, router.query, newSiteId);
+    // case 1: Update url with the selected site  param
+    // case 2: Update url with the selected site  param if both params are present in the url (i.e ploc, site)
+    if (selectedSite !== null || (requestedPlantLocation && requestedSite)) {
+      const pathname = `/${locale}/prd/${singleProject.slug}`;
+      const siteId = projectSites[selectedSite ?? 0].properties.id;
+      const updatedQueryParams = updateUrlWithParams(
+        router.asPath,
+        router.query,
+        siteId
+      );
+      pushWithShallow(pathname, updatedQueryParams);
+      return;
+    }
+
+    // Update url with the selected plant location  param
+    if (selectedPlantLocation) {
+      const pathname = `/${locale}/prd/${singleProject.slug}`;
+      const updatedQueryParams = { ploc: selectedPlantLocation.hid };
+      pushWithShallow(pathname, updatedQueryParams);
+    }
+  }, [
+    singleProject,
+    selectedSite,
+    selectedPlantLocation,
+    locale,
+    router.isReady,
+    requestedPlantLocation,
+    requestedSite,
+  ]);
 
```

Comment:
This useEffect is a bit confusing to me. What does this do, and can it be documented for later reference?